scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ClientHandshaker.java

```diff
@@ -152,25 +155,31 @@ public ClientHandshaker(DTLSSession session, RecordLayer recordLayer, SessionLis
 		this.publicKey = config.getPublicKey();
 		this.pskStore = config.getPskStore();
 		this.serverNameResolver = config.getServerNameResolver();
-		this.preferredCipherSuites = config.getSupportedCipherSuites();
+		this.preferredCipherSuites = Arrays.asList(config.getSupportedCipherSuites());
 		this.maxFragmentLengthCode = config.getMaxFragmentLengthCode();
 		this.supportedServerCertificateTypes = new ArrayList<>();
-		this.supportedServerCertificateTypes.add(CertificateType.RAW_PUBLIC_KEY);
-		if (rootCertificates != null && rootCertificates.length > 0) {
-			int index = config.isSendRawKey() ? 1 : 0;
-			this.supportedServerCertificateTypes.add(index, CertificateType.X_509);
-		}
-
 		this.supportedClientCertificateTypes = new ArrayList<>();
-		if (privateKey != null && publicKey != null) {
-			if (certificateChain == null || certificateChain.length == 0) {
-				this.supportedClientCertificateTypes.add(CertificateType.RAW_PUBLIC_KEY);
-			} else if (config.isSendRawKey()) {
-				this.supportedClientCertificateTypes.add(CertificateType.RAW_PUBLIC_KEY);
-				this.supportedClientCertificateTypes.add(CertificateType.X_509);
-			} else {
-				this.supportedClientCertificateTypes.add(CertificateType.X_509);
-				this.supportedClientCertificateTypes.add(CertificateType.RAW_PUBLIC_KEY);
+
+		// Currently scandium offers only ECC-based cipher suite, 
+		// but no other certificate-based CiperSuites.
+		// Correct this check, if other certificates are supported
+		if (CipherSuite.containsEccBasedCipherSuite(preferredCipherSuites)) {
```

Comment:
I do not think that this check should be based on the fact that an ECC based cipher suite is supported but based on the fact if any of the supported cipher suites require a certificate being exchanged. IMHO  we can use the existing `CipherSuite.requiresServerCertificateMessage()` method for doing such a check...